woocommerce/woocommerce
# Add an option to mini cart block to navigate directly to checkout.

### Changes proposed in this Pull Request:

Closes #45552

This adds a basic setting that changes behaviour of clicking the Mini-Cart icon. Instead of opening the drawer it will navigate to the checkout page when clicked. For this implementation I made the least assumptions possible to make a working implementation.

For example, in this implementation the behaviour is the same regardless of whether the cart is empty or not.

In my [original implementation](https://github.com/woocommerce/woocommerce/pull/51283/commits/439002b644acb9dc83a32c4bb923077d671a6022) I automatically disabled the "open drawer on add to cart" behaviour toggle when enabling this, but after testing this felt heavy handed and not necessary to force on users. The 2 behaviours can co-exist and while I don't know if anyone would want it I could not justify disabling the possibility.




### How to test the changes in this Pull Request:

1. Start with block theme enabled 
2. As a shopper user click the mini cart icon, by default it should open the drawer as before.
3. Now as merchant go to Appearance / Editor, and find the mini cart block on the page, click it and in the inspector panel enable the new toggle `Navigate to checkout when clicking the Mini-Cart, instead of opening the drawer.`
4. save changes
5. Now as shopper go to front end and click the mini cart and see you're taken straight to the checkout page



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add a setting to the Mini-Cart block to navigate to checkout on click instead of opening the drawer.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
